src/runtimes/node/bundlers/esbuild/bundler.ts

```diff
@@ -29,45 +29,48 @@ const RESOLVE_EXTENSIONS = ['.js', '.jsx', '.mjs', '.cjs', '.ts', '.tsx', '.mts'
 /**
  * Our own `includedFiles` syntax is slightly different from what esbuild expects as `externals`.
  *
- * Turns !node_modules/test/** into ./node_modules/test/*
+ * Turns !node_modules/test/** into test
+ * and !lang/en.* into ./lang/en.*.
+ * esbuild can't handle multiple globs in a single pattern, so we resolve them instead.
  */
-const includedFilesToEsbuildExternals = (
-  includedFiles: string[],
-  failOnDouble: boolean,
-  logger: Logger,
-  functionName: string,
-) =>
-  includedFiles
-    // only keep negated patterns
+const includedFilesToEsbuildExternals = async (includedFiles: string[], baseDir: string) => {
+  const exclusions = includedFiles
     .filter((pattern) => pattern.startsWith('!'))
-    // warn about invalid patterns
-    .filter((pattern) => {
-      const containsMultipleStars = pattern.indexOf('*') !== pattern.lastIndexOf('*')
+    .map((pattern) => pattern.slice(1))
+    // esbuild expects relative paths
+    .map((pattern) => `./${pattern}`)
+    // esbuild treats * the same as glob treats **, so this replacement is safe
+    .map((pattern) => pattern.replace(/\*\*/g, '*').replace(/\*(\\\*)+/g, '*'))
 
-      if (containsMultipleStars) {
-        const warning = `Found invalid pattern in includedFiles: ${pattern}. Only one * is allowed when using esbuild. See https://ntl.fyi/esbuild-double-glob for more.`
+  const result: string[] = []
 
-        if (failOnDouble) {
-          throw new FunctionBundlingUserError(warning, {
-            functionName,
-            runtime: RUNTIME.JAVASCRIPT,
-            bundler: NODE_BUNDLER.ESBUILD,
-          })
-        }
+  for (const pattern of exclusions) {
+    // esbuild expects modules to be passed in as module names, not paths
+    const nodeModulesMatch = pattern.match(/^\.\/node_modules\/(([^/]+)(\/[^/]*)?)*/)
```

Comment:
If I understand the logic correctly, something like `!node_modules/react-native/**/*` would have been transformed into `./node_modules/react-native/*` by the time it gets here?

If that's true, then I think the module name should come from the 3rd position in the match, not the 2nd?

https://jsbin.com/zamayipaka/edit?js,console

Reply:
Hmmm - I also want it to work for `@react-native/foo`. I believe the edit in https://github.com/netlify/zip-it-and-ship-it/pull/1557/commits/37450e26bca6235da2d73840d01e4d28463e2904 fixes that, at least it looks right when I dabble with it in https://jsbin.com/wanonatoro/edit?js,console :)